spec.bs

```diff
@@ -396,12 +396,16 @@ To <dfn>register an import map</dfn> given a [=string=] or a null |source text|,
   1. [=map/For each=] |scopePrefix| → |scopeImports| of |importMap|'s [=import map/scopes=],
     1. If |scopePrefix| is |baseURLString|, or if |scopePrefix| ends with U+002F (/) and |baseURLString| [=/starts with=] |scopePrefix|, then:
       1. Let |scopeImportsMatch| be the result of [=resolving an imports match=] given |normalizedSpecifier|, |scopeImports|, and |moduleMap|.
-      1. If |scopeImportsMatch| is not null, then return |scopeImportsMatch|.
+      1. If |scopeImportsMatch| is not null, then:
+        1. [=Validate the module script URL=] given |scopeImportsMatch|, |settingsObject|, and |baseURL|.
```

Comment:
I don't think this check will ever throw, since the import map will never return something that would fail the check.

Reply:
Right, but for now I'd like to apply `validate the module script URL` on every paths.
Currently the checks that ensures this `validate the module script URL` never failing are scattered across the spec and it's not so clear what is rejected or not. I hope we can simplify/centralize the checks, but after this PR.

Reply:
My motivation was to reduce the number of checks scattered throughout. But OK, in that case I'll add a comment to the source reminding us to remove this when we get a better grasp on things.